apps/frontend/src/libs/components/map-location-field/map-location-field.tsx

```diff
@@ -0,0 +1,65 @@
+import { type PointGeometry } from "@smartscapes/shared";
+import React from "react";
+import { type Control } from "react-hook-form";
+
+import { Icon, MapProvider } from "~/libs/components/components.js";
+import { combineClassNames } from "~/libs/helpers/combine-class-names.helper.js";
+import { useFormController } from "~/libs/hooks/hooks.js";
+import { type PointsOfInterestRequestDto } from "~/modules/points-of-interest/points-of-interest.js";
+
+import { MapLocationLogic } from "./map-location-logic/map-location-logic.js";
+import styles from "./styles.module.css";
+
+type Properties = {
+	control: Control<PointsOfInterestRequestDto>;
+	errorMessage?: string;
+	label?: string;
+	name: "location";
+};
+
+const MapLocationField = ({
+	control,
+	errorMessage,
+	label = "Location",
+	name,
+}: Properties): React.JSX.Element => {
```

Comment:
MapLocationField should now know nothing about the logical part where it's being used. It should be very generic and similar to usual inputs. 
```suggestion
type Properties<T extends FieldValues> = {
	control: Control<T, null>;
	errors: FieldErrors<T>;
	label: string;
	name: FieldPath<T>;
};

const MapLocationField = <T extends FieldValues>({
	control,
	errors,
	label,
	name,
}: Properties<T>): React.JSX.Element => {
```